File: demo/src/main/java/com/bytedance/scenedemo/lifecycle/LifeCycleDemoScene.java
package com.bytedance.scenedemo.lifecycle;

import android.os.Bundle;
import androidx.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import androidx.annotation.NonNull;
import com.bytedance.scene.group.GroupScene;
import com.bytedance.scenedemo.group.EmptyScene;

/**
 * Created by JiangQi on 8/28/18.
 */
public class LifeCycleDemoScene extends GroupScene {
    FrameLayout frameLayout;

    @NonNull
    @Override
    public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        frameLayout = new FrameLayout(getActivity());
        frameLayout.setId(View.generateViewId());

        return frameLayout;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        if (findSceneByTag("wo") == null) {
            add(frameLayout.getId(), EmptyScene.newInstance(0), "wo");
        }

        if (findSceneByTag("wo") == null) {
            add(frameLayout.getId(), EmptyScene.newInstance(0), "wo");
        }

//        beginTransaction();
//        if (findSceneByTag("wo") == null) {
//            add(frameLayout.getId(), EmptyScene.newInstance(0), "wo");
//        }
//
//        if (findSceneByTag("wo") == null) {
//            add(frameLayout.getId(), EmptyScene.newInstance(0), "wo");
//        }
//        commitTransaction();
    }

    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    public void onResume() {
        super.onResume();
    }
}

File: demo/src/main/java/com/bytedance/scenedemo/lifecycle/LifeCycleDemoScene.kt
package com.bytedance.scenedemo.lifecycle

import com.bytedance.scenedemo.lifecycle.EmptyScene.Companion.newInstance
import com.bytedance.scene.group.GroupScene
import android.widget.FrameLayout
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import com.bytedance.scene.Scene

/**
 * Created by JiangQi on 8/28/18.
 */
class LifeCycleDemoScene : GroupScene() {
    lateinit var frameLayout: FrameLayout
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        frameLayout = FrameLayout(requireActivity())
        frameLayout!!.id = View.generateViewId()
        return frameLayout
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if (findSceneByTag<Scene?>("wo") == null) {
            add(frameLayout!!.id, newInstance(0), "wo")
        }
        if (findSceneByTag<Scene?>("wo") == null) {
            add(frameLayout!!.id, newInstance(0), "wo")
        }

//        beginTransaction();
//        if (findSceneByTag("wo") == null) {
//            add(frameLayout.getId(), EmptyAppCompatScene.newInstance(0), "wo");
//        }
//
//        if (findSceneByTag("wo") == null) {
//            add(frameLayout.getId(), EmptyAppCompatScene.newInstance(0), "wo");
//        }
//        commitTransaction();
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onResume() {
        super.onResume()
    }
}